File: python-version/LC624/max_distance.py
from math import inf

class Solution:
    def maxDistance(self, arrays: list[list[int]]) -> int:
        ans = 0
        mn, mx = inf, -inf
        for a in arrays:
            ans = max(ans, a[-1] - mn, mx - a[0])
            mn = min(mn, a[0])
            mx = max(mx, a[-1])
        return ans

File: go-version/leetcode/LC624/max_distance.go
package lc624

import "math"

func maxDistance(arrays [][]int) int {
	mn, mx := math.MaxInt/2, math.MinInt/2
	ans := 0
	for _, arr := range arrays {
		first, last := arr[0], arr[len(arr)-1]
		ans = max(ans, mx-first, last-mn)
		mn = min(mn, first)
		mx = max(mx, last)
	}
	return ans
}
